### PR TITLE
Rewrite CLI in Go

### PR DESCRIPTION
**Pull request by lindluni**
This PR introduces a full rewrite of the `gh-token` CLI as a pure-Go `gh` CLI extension.

Of note is the choice of `urfave/cli` over `spf13/cobra` for the command package. Having spent years using `spf13/cobra` I prefer `urfave` for the simplicity due to the future features this extension itself actually requires, the Cobra command package provides much more functionality than we need.

While I pulled in `google/go-github`, I chose to use it for the types only, and not for using it as an API client. Using the `go-github` client pulls in many 3rd party dependencies, including the additional 3rd party `ghinstallations` library. All of this just increases the maintenance surface, so I chose to implement the http calls using pure Go, which reduces the maintenance burden of keeping the app's dependencies up to date.

To test this PR you can build and install the extension locally, in prod the extension will be built and pinned by Actions as a release:

```shell
gh extension remove token
git clone git@github.com:Link-/gh-token
cd gh-token
gh pr checkout 24
go build .
gh extension install .
```

### CODE REVIEW

**Comment by lindluni** [comment]
@Link- 

I'll ping you on Monday. I'd like to get some app credentials into the Actions Secret store for this repo for a test app so I can write e2e integration tests for the functionality.

Eventually I'll get to writing unit tests as well, but the scope of the CLI is small, so integration tests will be a better use of my dev cycles.

**Comment by Link-** [comment]
Fixes #24 - I'll review shortly

**Comment by lindluni** [comment]
Updated the original PR body, had the wrong commit SHA 🤦🏻

File: internal/generate_flags.go
Comment (Link-) [inline internal/generate_flags.go:12]: What are your thoughts on maintaining the same parameters as the bash version? I like yours better and yet I'm thinking about the folks who will be migrating, and/or already have this hooked to some other stuff. This might be a breaking change for them.

cs_2023-09-13_10-36-26

File: internal/installations.go
Comment (Link-) [inline internal/installations.go:113]: I didn't test this yet, and this is a rushed idea, but as it goes through the pages, will this spit out a single JSON array with the installation id objects or will it spit out multiple JSON arrays concatenated?

This was a bug in the GitHub CLI  that was later fixed, we might want to make sure we don't repeat it here.

**Problematic output**:
```
[{},{},{}][{},{},{}][{},{},{}]...
```

**Correct output:**
```
[{}, {}, {}, {}, {}, {}, {} ...]
```

File: images/gh-token.png
Comment (Link-) [inline images/gh-token.png:1]: Let's use: https://ray.so/ to re-generate this

**Review by Link- — CHANGES_REQUESTED** [review]
This is great, I still need to run some tests and we need to do the following before merging:

- [ ] Push out an announcement that we're migrating to Go
- [x] Update README.md to cover the new changes
- [ ] It'll be awesome to have an e2e test workflow (but this is a nice to have)

File: internal/generate_flags.go
Comment (lindluni) [reply internal/generate_flags.go:12]: I have an idea that I think I can do one better. `urfave/cli` will accept multiple aliases. I'm good with keeping the same names as long as we replace the `_` with `-` since that is pretty much standard. So the default for App ID would be `--app-id` but it would have an alias of `--app_id` and `-i` to preserve backwards compatibility. And during the release announcement we can mention that all flags containing underscores have been preserved for backwards compatibility and are deprecated and may be removed in the future.

I'd also not be opposed to the `_` flags being the default, and the `-` flags being the alias and making the same announcement that `_` flags are deprecated and users should begin using `-` ASAP as the `_` flags may be removed in the future.

Thoughts on which route you'd rather go?

Comment (Link-) [reply internal/generate_flags.go:12]: Great, I like it. Let's do it :)

Comment (lindluni) [reply internal/generate_flags.go]: I've added the old flags as aliases to preserve backwards compatibility, and updated the flags to match the names from `v1`

File: images/gh-token.png
Comment (lindluni) [reply images/gh-token.png:1]: I tried my hardest to match the format of the original one you had, but not sure if there is a paid version of ray.so that has more colors, but I couldn't get anywhere near as nice of a terminal as you had in the `v1` version.

Also, this tool is awesome, totally using this all the time now!

**Comment by lindluni** [comment]
> This is great, I still need to run some tests and we need to do the following before merging:
> 
> * [ ]  Push out an announcement that we're migrating to Go
> * [ ]  Update README.md to cover the new changes
> * [ ]  It'll be awesome to have an e2e test workflow (but this is a nice to have)



> This is great, I still need to run some tests and we need to do the following before merging:
> 
> * [ ]  Push out an announcement that we're migrating to Go
> * [ ]  Update README.md to cover the new changes
> * [ ]  It'll be awesome to have an e2e test workflow (but this is a nice to have)

I 💯 agree on the e2e tests, this should be simple to test all functions, I'll ping you in Slack with app credentials, if you can add them to the secret store I'll write the e2e tests tomorrow, I have some free cycles Friday afternoon.

**Comment by Link-** [comment]
@lindluni can you have a look at the recent changes? I think we're ready to merge this when this deadline hits: https://github.com/Link-/gh-token/discussions/26

**Comment by lindluni** [comment]
Love it!

It'd still be great if we could get some unit tests and then some E2E tests done in the future with an actual app, but that's a story for another PR, and maybe something someone in the community would be interested in helping with (we can open a `help-wanted` issue after the merge).

I do think we should create a tag for the current `main` as `v1.0.0` and then merge this and tag it as `v2.0.0` latest, so anyone who still needs the old code can grab it from the release.

**Review by Link- — APPROVED** [review]
Let's do this!

**Pull request merged**
